### DISCUSSION TITLE
ESP32 D1 and ESP32S follow up

### DISCUSSION THREAD

**Discussion by SunnyMoo** [awawa-dev/HyperSerialESP32#48] [General]
Hey Awawa !

Just to let you know, i managed to make the boards working.
First i figure that i didn't need relay pin, just bridging 3.3 pin to relay's positive and signal does the job.
So i used your vanilla firmwares and everything seems to work fine.

Note i couldn't make the custom compiled bins with relay pin work at all, they had funky behavior.
I also compiled a 4Mb for the D1 (CP2104) as vanilla is 2Mb but as soon as i push bauds beyond 2M i've got an "incorrect parameter" error in the logs.
Maybe there's something with github compilation that create issues or i'm doing something wrong ?

The D1 is working atm with 2Mb, just had a strange frozen led colors when i got my screen on and out of screensaver, but it's an hyperhdr issue i'll try to fix on my side before opening a ticket.

Best regards.

**Comment by awawa-dev** [comment]
Hi
If I remember correctly maximum speed of CP2104 chip is just 2Mb.

> Note i couldn't make the custom compiled bins with relay pin work at all, they had funky behavior.

Did you follow the advice on using relay pins? The relay is only active where there is a communication from HyperHDR.

**Comment by SunnyMoo** [comment]
The D1 board i have is a clone of the MH-ET Live listed here:
![image](https://github.com/awawa-dev/HyperSerialESP32/assets/50081326/5975e8d4-5c98-4dce-a22d-9ba708541829)
I assume that CP2104 should support 4Mb.

> Did you follow the advice on using relay pins?

Yes, the issue with firmware acting strangely was when i reused the previously compiled firmwares while not using relay pin anymore.

**Reply by awawa-dev** [reply]
I admit that the description is not very precise, but 4Mb applies to ch9102x.

>> Did you follow the advice on using relay pins?
>
> Yes, the issue with firmware acting strangely was when i reused the previously compiled firmwares while not using relay pin anymore.

it's about HyperHDR configuration, it's explained on the main page.

**Comment by SunnyMoo** [comment]
> it's about HyperHDR configuration, it's explained on the main page.

What I mean is the bin compiled with relay pin were unstable and not reacting as wanted on hyperhdr commands eg turn on leds result in random led state 😅

But whatever, now it's working with vanilla firmwares even if i'm a bit confused about the CP2104 doing 2Mb only but anyway i can't get CH9102 or S2 mini, so will do with the 2mb.

Thank you !

**Reply by awawa-dev** [reply]
Not using recommended 'continues output' option can cause such effect: normally smoothing doesn't send data to LED device if LED colors does not change e.g. static color or scene => HyperSerial sees no communication and disables the power relay and LED. Then the comunicaion is resumed and power relay suddenly is on again. Handshake also restarts LEDs and additionally the ESP device when HyperHDR enables/disables Adalight device.
